plugin/contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -632,12 +632,18 @@ func (k *deleter) deleteOne(pod *Pod) error {
 }
 
 // Create creates a scheduler plugin and all supporting background functions.
-func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux) *PluginConfig {
+func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux,
+	makePodsWatcher func() *cache.ListWatch) *PluginConfig {
```

Comment:
You shouldn't need to make `createAllPodsLW` public. `NewDefaultPluginConfig` would just use it internally and delegate to `NewPluginConfig`.